File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        n=len(nums)
        remainder_map={0:-1}
        prefix_sum=0
        for i in range(n):
            prefix_sum+=nums[i]
            remainder=prefix_sum%k
            if remainder in remainder_map:
                if i-remainder_map[remainder]>=2:
                    return True 
            else :
                remainder_map[remainder]=i
        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
          unordered_map<int, int> seen;
        seen[0] = -1;

        int prefix = 0;

        for (int i = 0; i < nums.size(); i++) {
            prefix += nums[i];

            int remainder = prefix % k;

            if (seen.find(remainder) != seen.end()) {
                if (i - seen[remainder] >= 2)
                    return true;
            }
            else {
                seen[remainder] = i;
            }
        }

        return false;
    }
};